### ISSUE TITLE
Inconsistent comment type in C/C++

### ISSUE THREAD

**Issue by Ram-Z** [tomtom/tcomment_vim#222]
When using `gcc` to comment a whole line, the comment type is inconsistent sometimes using `/* */` and sometimes `// `.

Given the following code,
```cpp
int a = 1;
auto a = 1;
```
running `gcc` on each line individually results in
```cpp
/* int a = 1; */
// auto a = 1;
```

This seems to be a regression. It only started happening after I upgraded my plugins recently.

**Comment by Ram-Z** [comment]
That works for me.

**Comment by tomtom** [comment]
I cannot reproduce this problem. Could you please post the result of `:echo synIDattr(synID(line("."), col("."), 1), "name")` with the cursor placed at int and auto?

**Comment by Ram-Z** [comment]
`int` is "cType" and `auto` is "cppType".

**Comment by Bothari** [comment]
I started getting this issue as well when I updated my plugins. I have a simple and confusing test case. File named "foo.cpp", filetype is detected as cpp. Entire contents of file is as follows:
```
{
	if (foo)
		return bar;
}

if (foo)
	return bar;
```
Doing the echo you posted above it returns ```cStatement``` for both ```return``` lines, and ```cConditional``` for both ```if()``` lines. I would expect it to do C++ style ```//``` comments for all lines. Commenting the lines individually with tcomment results in the following:
```
{
	// if (foo)
		/* return bar; */
}

/* if (foo) */
	/* return bar; */
```
Confirmed that disabling ```g:tcomment#filetype#guess``` appears to fix the issue.
